File: LeetCode/LeetCode Practice Problems/1816- Truncate Sentence.cpp
class Solution {
public:
    string truncateSentence(string s, int k) {
        s += " ";
        string ans = "";
        string cur = "";
        int n = s.length();
        for(int i=0;i<n && k>0;i++) {
            if(s[i] != ' ') {
                cur += s[i];
            } else {
                ans += cur + ((k == 1)?"":" ");
                cur = "";
                k--;
            }
        }
        
        return ans;
        
    }
};

File: LeetCode/LeetCode Practice Problems/1816-Truncate Sentence.java
class Solution {
    public String truncateSentence(String s, int k) {
        s += " ";
        String ans = "";
        String cur = "";
        
        for(char c: s.toCharArray()) {
            if(c != ' ') {
                cur += c;
            } else {
                k--;
                if(ans.length() != 0) ans += " ";
                ans += cur;
                cur = "";
            }
            
            if(k == 0) break;
        }
        
        return ans;
    }
}
